Fix formatting

rodrigoprimo opened the pull request:
[MAILPOET-5232]

[MAILPOET-5232]: https://mailpoet.atlassian.net/browse/MAILPOET-5232?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

rodrigoprimo committed:
Sanitize name and description when creating a segment

This commits adds the sanitization right before the data is added to the
database (\MailPoet\Segments\SegmentsRepository::createOrUpdate()) and
removes the sanitization from
\MailPoet\Segments\DynamicSegments\SegmentSaveController::save() to
avoid sanitizing twice. save() calls createOrUpdate().

Before this commit, we were sanitizing the name and description of
dynamic segments but not regular segments.

[MAILPOET-5232]

rodrigoprimo committed:
Always escape segment name and description in the segment list page

Please note that this change might impact some users as before we were
displaying HTML tags in the segment description but now the < will be
convert to an HTML entity.

[MAILPOET-5232]

rodrigoprimo committed:
Escape the option name when displaying items in a list in a form

This should prevent an XSS vector both in the segments list in the page
to edit a subscriber and in the segments lists in the page to send a
newsletter.

[MAILPOET-5232]

triple0t commented on mailpoet/lib/Segments/DynamicSegments/SegmentSaveController.php:
```diff
@@ -31,8 +31,8 @@ public function __construct(
     $id = isset($data['id']) ? (int)$data['id'] : null;
-    $name = isset($data['name']) ? sanitize_text_field($data['name']) : '';
-    $description = isset($data['description']) ? sanitize_textarea_field($data['description']) : '';
+    $name = $data['name'] ?? '';
+    $description = $data['description'] ?? '';
```
If we are doing this here, You might also want to update this section: https://github.com/mailpoet/mailpoet/blob/0bbd65ac61f009ea234ddf59baf5b10e507ca11d/mailpoet/lib/API/MP/v1/Segments.php#L45-L46

rodrigoprimo replied:
Thanks for the review, @triple0t. I will follow on a DM.

triple0t reviewed (commented):
LGTM

rodrigoprimo committed:
Handle sanitization as early as possible per best practices

Per WP security best practices, sanitization should be handled as early
as possible. So this commit move updates the calls to sanitize the
segment name and description to the part of the code where the user
input is first processed, instead of when the data is saved to the
database.

[MAILPOET-5232]

rodrigoprimo committed:
Remove PHPStan ignored error as it is not happening anymore

[MAILPOET-5232]

brezocordero reviewed (approved):
The changes look good.
